tests/test_bit_manip.cc

```diff
@@ -76,4 +76,28 @@ TEMPLATE_TEST_CASE("Access MSB", "[bit_manip]",
     REQUIRE(gul14::bit_test(x, msb - 1) == false);
 }
 
+TEST_CASE("is_little_endian(), is_big_endian()", "[bit_manip]")
+{
+    REQUIRE(gul14::is_little_endian() != gul14::is_big_endian());
+
+    const std::uint32_t val = 0x01'02'03'04;
+    const void* ptr = &val;
+    const char* chars = static_cast<const char*>(ptr);
+
+    if (gul14::is_little_endian())
+    {
+        REQUIRE(chars[0] == 4);
```

Comment:
The lifetime of the char-wide number at `chars[0]` never began -- with C++17 we have `std::launder`, here it is implementation defined (but there are no implementations known to not work as expected).

Iirc there is no way to do this at all in a standard conforming way in <= C++14. Except using C instead.

(I believe :grimacing: )

Reply:
What we are doing is by definition *implementation defined*, so I fail to see the problem here?

Reply:
Hmm, I guess accessing something (via a pointer) which lifetime has not started is an ill-formed program iirc.
It does work (because of the C legacy), and there is no way to program something like this in a conforming way prior to C++17, but anyhow.

> I fail to see the problem here?

There is no problem. But then we have code in GUL that meticulously handles memory alignment although there would be no problem without it. We make stuff noexcept for no real benefit but just because it seems correct. And here we just hand-wave over obvious standard violations.
And for which reasons? Just that one or two doocs libraries can keep their old code and not have the `endian` problem there?

The more I think about it the more I dislike having the function in GUL at all, sorry :disappointed: